plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/storage/TestTrinoHudiStorage.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hudi.storage;
+
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.local.LocalFileSystemFactory;
+import io.trino.spi.block.TestingSession;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.io.storage.TestHoodieStorageBase;
+import org.apache.hudi.io.util.IOUtils;
+import org.apache.hudi.storage.HoodieStorage;
+import org.apache.hudi.storage.StoragePath;
+import org.apache.hudi.storage.StoragePathInfo;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestTrinoHudiStorage
+        extends TestHoodieStorageBase
+{
+    private static final byte[] EMPTY_BYTES = new byte[] {};
+
+    private static TrinoFileSystem fileSystem;
+
+    @BeforeAll
+    static void setUp()
+    {
+        fileSystem = new LocalFileSystemFactory(Path.of("/tmp"))
+                .create(TestingSession.SESSION);
+    }
+
+    @Override
+    protected HoodieStorage getStorage(Object fileSystem, Object config)
+    {
+        return new TrinoHudiStorage((TrinoFileSystem) fileSystem, (TrinoStorageConfiguration) config);
+    }
+
+    @Override
+    protected TrinoFileSystem getFileSystem(Object config)
+    {
+        return fileSystem;
+    }
+
+    @Override
+    protected TrinoStorageConfiguration getConf()
+    {
+        return new TrinoStorageConfiguration();
+    }
+
+    @Override
+    protected String getTempDir()
+    {
+        return "local:" + this.tempDir.toUri().getPath();
+    }
+
+    @Override
+    @Test
+    public void testGetUri()
+            throws URISyntaxException
+    {
+        assertThat(getStorage().getUri()).isEqualTo(new URI(""));
+    }
+
+    @Override
```

Comment:
Please leave code comments why this test is overridden. Same for others.

Reply:
Added.